### PR TITLE
Simplify standalone collector code

### PR DESCRIPTION
**Pull request by srikartati**
Modify standalone collector code to make it extendable in the future.

### CODE REVIEW

**Review by zyiou — DISMISSED** [review]
LGTM

File: cmd/collector/collector.go [hunk 1]
```diff
@@ -60,6 +60,19 @@ func initLoggingToFile(fs *pflag.FlagSet) {
...
+	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+
+	for {
+		select {
+		case <-signalCh:
+			close(stopCh)
+			return
+		}
+	}
+}
```
Comment (shihhaoli) [inline cmd/collector/collector.go:74]: Just curious, if we have only one case, do we still need to use select?
For example, does the following suit our need?

```
func signalHandler(stopCh chan struct{}) {
	signalCh := make(chan os.Signal, 1)
	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
	<-signalCh
	close(stopCh)
}
```
Thanks.

Comment (srikartati) [reply cmd/collector/collector.go:81]: Yes, you are right. We can simplify the code.
Taking inspiration from Antrea applications, I defined signalCh with capacity 2 to handle the case if a second interrupt/termination comes through. In that case, program exits forcefully.

**Comment by srikartati** [comment]
> So if the second signal was not received, this function will not exit normally?

I think you meant the go routine; the function returns immediately after starting the go routine. Yes that go routine abruptly stops when the main exits gracefully. Main does not need to wait for this signal handler go routine to finish.

File: cmd/collector/collector.go (hunk 1 above)
Comment (shihhaoli) [reply cmd/collector/collector.go:81]: Yes, I meant the go routine. Thanks for the explanation.

File: cmd/collector/collector.go [hunk 2]
```diff
@@ -60,6 +60,26 @@ func initLoggingToFile(fs *pflag.FlagSet) {
...
+		<-signalCh
+		close(stopCh)
+		<-signalCh
+		klog.Warning("Received second signal, will force exit")
+		logs.FlushLogs()
+		os.Exit(1)
+	}()
+
+	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	return
```
Comment (shihhaoli) [inline cmd/collector/collector.go:80]: Looks like the return is not needed.

**Review by shihhaoli — APPROVED** [review]
LGTM.

**Review by shihhaoli — APPROVED** [review]

**Pull request merged**